src/python/doc/wasserstein_distance_user.rst

```diff
@@ -100,6 +100,57 @@ The output is:
     point 2 in dgm1 is matched to the diagonal
     point 1 in dgm2 is matched to the diagonal
 
+
+Entropic regularization
+***********************
+
+When using the Optimal Transport based version of wasserstein distances, 
+we also allow for the use of entropic regularization of optimal transport distance via the parameter ``reg``.
+If ``reg > 0``, we solve 
+:math:`\min_P \sum P_{ij} M_{ij} + reg * h(P)`
+under marginal constraints on :math:`P` , which can be understood as a generalized matching between the diagrams.
+:math:`P_{ij}` denotes the quantity of mass from the :math:`i`-th points in the first diagram that is
+transported to the :math:`j`-th point in the second diagram. Last row and column of :math:`P` denotes the diagonal
+by convention.
+Here :math:`M` is a matrix encoding the pairwises distances between the points of the two diagrams along with
+distances from points to the diagonal, and :math:`h(P) = \sum P_{ij} \log(P_{ij})` is the entropic regularization term.
+
+If ``matching=True``, we return both the estimated transport cost :math:`\sum P_{ij} C_{ij}` and the computed :math:`P`, 
+otherwise we simply return the estimated transport cost.
+
+The closer ``reg`` is to 0, the closer the transport cost will be from the exact distance between the diagrams.
+The larger the regularization parameter, the faster the computation.
+Note however that taking too low values for ``reg`` can run into numerical issues.
```

Comment:
What matters for numerical stability is the ratio `M / reg`, where M is the pairwise distance matrix between points (including the diagonal), and essentially because you compute at some point `exp(-M/reg)` (term-wise exponentiation), and you don't want it to be 0. So numerical issues arise when `max(M) / reg \sim 1000`. 

In this implementation, you indeed have a control which (theoretically) scales linearly with the number of points and with the entries of `M`. Roughly speaking, the more points you have and the larger the coordinate, the smaller you might want to pick `reg` to reach a fixed _additive_ error (it is not a surprise: the more points / the larger the coordinates, the larger the exact distance. Although I don't know how to prove it, I think the relative error is less impacted). Note also that these "linear additive errors" are worst-case bounds, the error made in practice is in general much smaller. 

I don't know a heuristic to pick a value for `reg`. From my intuition, this should just be considered as another notion of similarity between persistence diagrams, which might have various benefits (of interest, the optimal transport plan is more stable/differentiable). 
I'd say it is somewhat similar to using the `order=p` metric instead of the bottleneck distance to compare persistence diagrams. Although those are linked in the regime p-->infty, in practice you don't really care about this regime and use p=1 or p=2.

Reply:
Some discussion like that in the doc could be useful. At least knowing that reg is homogeneous to a distance.